NBXplorer/Repository.cs

```diff
@@ -809,6 +809,30 @@ private static SavedTransaction ToSavedTransaction(Network network, ReadOnlyMemo
 			t.Transaction.PrecomputeHash(true, false);
 			return t;
 		}
+		public async Task<MultiValueDictionary<OutPoint, KeyPathInformation>> GetKeyInformations(OutPoint[] outpoints)
+		{
+			// This is a unoptimized code, written so that it just gives the expected output.
+			// Should be replaced later on by a code like in GetKeyInformations(Script[] scripts)
+			MultiValueDictionary<OutPoint, KeyPathInformation> result = new MultiValueDictionary<OutPoint, KeyPathInformation>();
+			Dictionary<OutPoint, Script> outpointToScript = new Dictionary<OutPoint, Script>();
+			var scripts = new List<Script>();
+			foreach (var outpoint in outpoints)
+			{
+				var prevOutTxn = getTransaction(outpoint.Hash);
```

Comment:
This is sadly not possible, the server would never been able to keep up with the blockchain if we needed 1 roundtrip for each outpoint.

Reply:
Initially I was also against the idea of making a round trip for each outpoint. But I could not find a function through which I could collectively query all the outpoints for the corresponding txns. This part of code was a sort of place holder for such a function.

Reply:
not only this, but txindex is not mandatory for nbxplorer.

Reply:
What do you mean by txindex not mandatory for nbxplorer?

Reply:
Is it that we do not have to query for every txid we encounter, only the saved ones.